### PR TITLE
Use buildAndSignTransactionNew in quitStakePool

### PR DESCRIPTION
**Pull request by Unisay**
Wallet uses `buildAndSignTransactionNew` when quitting a stake pool.

### Issue Number

ADP-2267

### CODE REVIEW

**Review by Anviking — APPROVED** [review]
lgtm! Modulo we should write change state to DB before we merge (e.g. #3691)

Some minor thoughts
- Should we add `atomicallyWithHandler` like one `postTransactionOld`? 🤔 
- I'm curious how this affects performance. But `postTransactionOld` is likely the endpoint where this would matter though / easier to test.

**Comment by Unisay** [comment]
Closed in favor of https://github.com/input-output-hk/cardano-wallet/pull/3719

**Pull request closed without merging**
